+nigeLab/@Block/subsref.m

```diff
@@ -99,108 +97,127 @@
-            % Return all samples of those channels
-            ch = subs2idx(S.subs{1});
-            varargout{nargo} = nan(numel(ch),blockObj.Samples);
-            
-            for iCh = 1:numel(ch)
-               varargout{nargo}(iCh,:) = ...
-                  blockObj.Channels(ch(iCh)).(shortField).data(:);
-            end
-            return;
-            
-         case 2
-            % In this case, {'raw',[1,5,19],1:100} or {[1,5,19],1:100} for
-            % example. The sample indexing is also specified.
-            ch = subs2idx(S.subs{1});
-            samples = subs2idx(S.subs{2});
-            varargout{nargo} = nan(numel(ch),numel(samples));
-            for iCh = 1:numel(ch)
-               varargout{nargo}(iCh,:) = ...
-                  blockObj.Channels(ch(iCh)).(shortField).data(samples);
-            end
-            return;
-
-         otherwise
-            error(['nigeLab:' mfilename ':tooManySubscripts'],...
-               'Too many subscripts (%g) used for shortcut reference.',nargsi);
+      TargetObj = subsref(blockObj,substruct(substractInArgs{:}));
+      
+      subs = subs(end);     % we imposed only 2 levels of depth, this means subs has to be only one after this
+      if iscell(subs{1}),subs = subs{end};end
+      shrtStruct.subfields(1:ii) = [];
+      shrtStruct.indexable(1:ii) = [];
+      
+      
+      % Let's create the subsref structs
+      % Again we need to gahter the substruct info from the shortcut
+      
+      substractInArgs = {};
+      for ii = 1:numel(shrtStruct.subfields)
+          substractInArgs = [substractInArgs, {'.'} ,shrtStruct.subfields(ii)];
+          if shrtStruct.indexable(ii)
+              substractInArgs = [substractInArgs, {'()'}];
+              break;
+          else
+              ... Do nothig, go on with the loop
+          end
       end
+      
+%        but this time we add the indexes after the () in a different way
+      for jj = 1:numel(subs)
+          if ~checkInputArgsCoherence(numel(TargetObj),subs{jj})
+              % check that dimensions are coherent between inputs
+              error(['nigeLab:' mfilename ':badSubscriptReference'],...
+                   'Dimensions of arrays being concatenated are not consistent.');
+          end
+          
+          switch checkSamplesInput(subs{jj})
+              % habldes cell, not cell nonsense with substruct
+              case {'numeric','semicolon'}                 
+                   s(jj,:) = substruct(substractInArgs{:},subs(jj));
+              case 'cell'
+                  s(jj,:) = substruct(substractInArgs{:},subs{jj});
+              otherwise
+                  error(['nigeLab:' mfilename 'badSubscript'],...
+                       'Bad subscript number 3. %ss are not supported.',class(subs))
+          end
+         
+      end
+      index = ones(1,length(TargetObj)).*(1:size(s,1));
+      varargout = {arrayfun(@(x,idx) subsref(x,s(idx,:)),TargetObj,index,'UniformOutput',false)};
```

Comment:
One thing that's somewhat minor but I could see it being problematic in the future: by convention we have been trying to use row arrays for everything since Matlab `for` loops iterate on individual elements (columns) of a row vector. So we should be consistent with how the vectors are returned: currently `subsref` is returning column vectors--can we switch it so it automatically returns `row` vectors?

Reply:
This has been addressed with most recent `Push`